sites/all/modules/hr/hr_datasets/js/hr_datasets.js

```diff
@@ -142,12 +149,36 @@ Drupal.behaviors.hrDatasets = {
         },
 
         searchByTitle: function(event) {
-          var val = $('#search').val();
+          if(event.keyCode == 13) {
+            var val = $('#search').val();
+            if (val != '') {
+              this.DatasetsList.params.title = val;
```

Comment:
you're doing this if val not empty add value, else remove value then navigate several times, I think we should have an helper function here.